std/functional.d

```diff
@@ -260,6 +210,129 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by $(D argc). If no explicit
+        $(D argc) is given, it is inferred to be $(D parmNames.length).)
+    $(LI Explicit names for the function parameters may instead be given as
+        $(D parmNames). If more than $(D argc) names are supplied, the
+        extras will be ignored.)
+    $(LI If no $(D parmNames) are given, the letters ($(D a) through $(D z))
+        are used to name up to the first 26 function parameters.)
+    $(LI $(D args[index]) can always be used to access any parameter by index,
+        regardless of whether it has been given a name.)
+)
+*/
+template naryFun(alias fun, size_t argc, parmNames...)
```

Comment:
I don't get it. As far as I can see, if you pass anything to `parmNames`, you always want to pass `parmNames.length` to `argc`.